src/node_crypto.cc

```diff
@@ -877,14 +885,27 @@ void SecureContext::AddRootCerts(const FunctionCallbackInfo<Value>& args) {
         // We do not call back into JS after this line anyway, so ignoring
         // the return value of ProcessEmitWarning does not affect how a
         // possible exception would be propagated.
-        ProcessEmitWarning(sc->env(),
+        ProcessEmitWarning(env,
                            "Ignoring extra certs from `%s`, "
                            "load failed: %s\n",
                            extra_root_certs_file.c_str(),
                            ERR_error_string(err, nullptr));
+      } else {
+        extra_root_certs_file_loaded = true;
       }
     }
   }
+}
+
+
+void SecureContext::AddRootCerts(const FunctionCallbackInfo<Value>& args) {
+  SecureContext* sc;
+  ASSIGN_OR_RETURN_UNWRAP(&sc, args.Holder());
+  ClearErrorOnReturn clear_error_on_return;
+
+  if (root_cert_store == nullptr) {
```

Comment:
With the new logic could this be replaced with an `CHECK_NOT_NULL(root_cert_store)`

Reply:
No. `CHECK_NOT_NULL` might throw so that it is not expected here. But thanks for reminding me of using `CHECK_NOT_NULL`.

Reply:
If we do initialize `root_cert_store` very early in the process start, what are the possible conditions that `root_cert_store` can be `nullptr` and still be valid.
(I was looking for code doing `root_cert_store = ...` and found none)

Reply:
I have moved the `LoadExtraRootCertsFile` into `UseExtraCaCerts` so that `root_cert_store` may be `nullptr` when we don't set `NODE_EXTRA_CA_CERTS`.